bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -775,4 +814,122 @@ private void updateThingsScheduled(int delayMilliSeconds) {
             scheduledUpdateTask = scheduler.schedule(() -> updateThingsNow(), delayMilliSeconds, TimeUnit.MILLISECONDS);
         }
     }
+
+    /**
+     * Load the set of automation script ids.
+     */
+    private void loadAutomationScriptIds() {
+        try {
+            synchronized (this) {
```

Comment:
It would be better to synchronize on some private object instead of `this`, because you never know what other logic is also synchronizing on the object instance.

Reply:
> you never know 

Hmm. I do know .. since I wrote it all. :) The synchronize on this is called inline so there is no risk of cross locking. However I can change it to synchronize on the enclosing methods instead.

Reply:
I didn't check if there are already other synchronizations on `this`, but there is also the risk of this being introduced in the future (by someone else 😉), so it would be cleaner to avoid it altogether. You can only use this approach once anyway, so if you need another lock for a different purpose, you would have to use another object anyway.

Reply:
> I didn't check if there are already other synchronizations on this

The 'initialize()` and `dispose()` methods do `synchronize (this)` however I will not change that. But I will change the other newly introduced `synchronize (this)` calls.